shell/platform/tizen/key_event_handler.cc

```diff
@@ -44,14 +44,16 @@ Eina_Bool KeyEventHandler::OnKey(void* data, int type, void* event) {
           key->keyname, key->modifiers, is_down);
 
   if (engine->text_input_channel) {
-    if (is_down) {
+    if (is_down && engine->text_input_channel->IsSoftwareKeyboardShowing()) {
       engine->text_input_channel->OnKeyDown(key);
-    }
-    if (engine->text_input_channel->IsSoftwareKeyboardShowing()) {
       return ECORE_CALLBACK_PASS_ON;
     }
   }
```

Comment:
The text input channel only handles the key event when the software keyboard is shown?

Reply:
Well... strictly, I should say no, If the ```TextInputModel``` is valid, it can handle key-event. 
It is created when ```TextInputChannel``` receives  ```TextInput.setClient``` as a message.
(The message for showing the keyboard is ```TextInput.show```.)

Reply:
How about [this](https://github.com/flutter-tizen/engine/pull/142/commits/07d2c1a484ade11159ada4598b60ce29cadfa6cd)?

Reply:
Mmmm. You have to carefully see how the original logic looks like. It skips sending a key event to `key_event_channel` only when the software keyboard is shown.

Reply:
I think... it's changed a bit, but the behavior hasn't changed much.